### ISSUE TITLE
Tombstone cleanup panicked after compressing vectors with PQ

### ISSUE THREAD

**Issue by AlithAnar** [weaviate/weaviate#4337]
### How to reproduce this bug?

- set PQ as enabled by updating the schema
- restart kubernetes STS (it restarts weaviate pod)
- wait until PQ compression process kicks in

### What is the expected behavior?

The process should end and we should see log "Switching to compressed vectors"

### What is the actual behavior?

We have a following panic:

```{"level":"error","msg":"class AISkillV2: tombstone cleanup panicked","panic":"runtime error: index out of range [0] with length 0","time":"2024-02-28T11:18:25Z"}
goroutine 29759 [running]:
runtime/debug.Stack()
	/usr/local/go/src/runtime/debug/stack.go:24 +0x5e
runtime/debug.PrintStack()
	/usr/local/go/src/runtime/debug/stack.go:16 +0x13
github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw.(*hnsw).cleanUpTombstonedNodes.func1()
	/go/src/github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw/delete.go:214 +0xca
panic({0x189b780?, 0xc0034fa2b8?})
	/usr/local/go/src/runtime/panic.go:920 +0x270
github.com/weaviate/weaviate/adapters/repos/db/vector/ssdhelpers.ExtractCode8(...)
	/go/src/github.com/weaviate/weaviate/adapters/repos/db/vector/ssdhelpers/product_quantization.go:251
github.com/weaviate/weaviate/adapters/repos/db/vector/ssdhelpers.(*ProductQuantizer).Decode(0xc003550580, {0x0, 0x0, 0xc?})
	/go/src/github.com/weaviate/weaviate/adapters/repos/db/vector/ssdhelpers/product_quantization.go:383 +0x186
github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw.(*hnsw).reassignNeighbor(0xc00422e000, 0x5b08, {0x1db38d8, 0xc00288c2f8}, 0x2c49?)
	/go/src/github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw/delete.go:332 +0x293
github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw.(*hnsw).reassignNeighborsOf(0xc00422e000, {0x1db38d8, 0xc00288c2f8}, 0xc000079b68?)
	/go/src/github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw/delete.go:297 +0xe7
github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw.(*hnsw).cleanUpTombstonedNodes(0xc00422e000, 0xc0027f38e0)
	/go/src/github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw/delete.go:236 +0x1f8
github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw.(*hnsw).tombstoneCleanup(0xc00422e000, 0x412b01?)
	/go/src/github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw/startup.go:179 +0x2a
github.com/weaviate/weaviate/entities/cyclemanager.(*cycleCallbackGroup).cycleCallbackSequential.func1(0xc00007d140, 0xc003764a20, 0x0?, 0xc0024a0450, 0x0, 0xc000025e2b)
	/go/src/github.com/weaviate/weaviate/entities/cyclemanager/cyclecallbackgroup.go:139 +0xe2
github.com/weaviate/weaviate/entities/cyclemanager.(*cycleCallbackGroup).cycleCallbackSequential(0xc00007d140, 0xc0024a0450)
	/go/src/github.com/weaviate/weaviate/entities/cyclemanager/cyclecallbackgroup.go:164 +0x289
github.com/weaviate/weaviate/entities/cyclemanager.(*cycleCallbackGroup).CycleCallback(0xc0028e56e8?, 0x412b85?)
	/go/src/github.com/weaviate/weaviate/entities/cyclemanager/cyclecallbackgroup.go:89 +0x1d
github.com/weaviate/weaviate/entities/cyclemanager.(*cycleCallbackGroup).cycleCallbackParallel.func1.1(0x1da90e0?, 0xc00007d1a0, 0x0?, 0xc16fe37e4e529c3a?, 0xc002dc3550, 0xc002dc3518)
	/go/src/github.com/weaviate/weaviate/entities/cyclemanager/cyclecallbackgroup.go:208 +0x84
github.com/weaviate/weaviate/entities/cyclemanager.(*cycleCallbackGroup).cycleCallbackParallel.func1()
	/go/src/github.com/weaviate/weaviate/entities/cyclemanager/cyclecallbackgroup.go:221 +0x26a
created by github.com/weaviate/weaviate/entities/cyclemanager.(*cycleCallbackGroup).cycleCallbackParallel in goroutine 188
	/go/src/github.com/weaviate/weaviate/entities/cyclemanager/cyclecallbackgroup.go:179 +0xa6
```

### Supporting information

Embedding model: gtr-t5-large

1 weaviate replica with 1 shard

Schema:
```
{
  "classes": [
    {
      "class": "AISkillV2",
      "description": "Contains chunked text content of a skills",
      "invertedIndexConfig": {
        "bm25": { "b": 0.75, "k1": 1.2 },
        "cleanupIntervalSeconds": 60,
        "indexNullState": true,
        "stopwords": { "additions": null, "preset": "en", "removals": null }
      },
      "moduleConfig": {
        "text2vec-transformers": {
          "poolingStrategy": "masked_mean",
          "vectorizeClassName": false
        }
      },
      "multiTenancyConfig": { "enabled": false },
      "properties": [
        {
          "dataType": ["text"],
          "description": "Vectorized chunk of text of skill",
          "indexFilterable": true,
          "indexSearchable": true,
          "moduleConfig": {
            "text2vec-transformers": {
              "skip": false,
              "vectorizePropertyName": false
            }
          },
          "name": "content",
          "tokenization": "word"
        },
        {
          "dataType": ["text"],
          "description": "URL address of page containing this chunk of text",
          "indexFilterable": true,
          "indexSearchable": true,
          "moduleConfig": {
            "text2vec-transformers": {
              "skip": true,
              "vectorizePropertyName": false
            }
          },
          "name": "url",
          "tokenization": "word"
        },
        {
          "dataType": ["uuid"],
          "description": "Id of organization issued by GlobalAccounts",
          "indexFilterable": true,
          "indexSearchable": false,
          "moduleConfig": {
            "text2vec-transformers": {
              "skip": true,
              "vectorizePropertyName": false
            }
          },
          "name": "organization_id"
        },
        {
          "dataType": ["uuid"],
          "description": "Id of skill containing this text chunks",
          "indexFilterable": true,
          "indexSearchable": false,
          "moduleConfig": {
            "text2vec-transformers": {
              "skip": true,
              "vectorizePropertyName": false
            }
          },
          "name": "skill_id"
        },
        {
          "dataType": ["uuid"],
          "description": "Specific revision of the skill",
          "indexFilterable": true,
          "indexSearchable": false,
          "moduleConfig": {
            "text2vec-transformers": {
              "skip": true,
              "vectorizePropertyName": false
            }
          },
          "name": "skill_revision"
        },
        {
          "dataType": ["date"],
          "description": "Last time this record was verified to be a part of valid skill revision",
          "indexFilterable": true,
          "indexSearchable": false,
          "moduleConfig": {
            "text2vec-transformers": {
              "skip": true,
              "vectorizePropertyName": false
            }
          },
          "name": "verified_at"
        }
      ],
      "replicationConfig": { "factor": 1 },
      "shardingConfig": {
        "virtualPerPhysical": 128,
        "desiredCount": 1,
        "actualCount": 1,
        "desiredVirtualCount": 128,
        "actualVirtualCount": 128,
        "key": "_id",
        "strategy": "hash",
        "function": "murmur3"
      },
      "vectorIndexConfig": {
        "skip": false,
        "cleanupIntervalSeconds": 300,
        "maxConnections": 64,
        "efConstruction": 128,
        "ef": -1,
        "dynamicEfMin": 100,
        "dynamicEfMax": 500,
        "dynamicEfFactor": 8,
        "vectorCacheMaxObjects": 1000000000000,
        "flatSearchCutoff": 40000,
        "distance": "cosine",
        "pq": {
          "enabled": true,
          "bitCompression": false,
          "segments": 192,
          "centroids": 256,
          "trainingLimit": 100000,
          "encoder": { "type": "kmeans", "distribution": "log-normal" }
        }
      },
      "vectorIndexType": "hnsw",
      "vectorizer": "text2vec-transformers"
    }
  ]
}

```

### Server Version

1.23.9

### Code of Conduct

- [X] I have read and agree to the Weaviate's [Contributor Guide](https://weaviate.io/developers/contributor-guide) and [Code of Conduct](https://weaviate.io/service/code-of-conduct)

**Comment by rthiiyer82** [comment]
@AlithAnar : Thank you fr reporting this issue. I can confirm that the issue is reproducible in 1.24 release as well. We are working on a fix and will confirm the release when available.

**Comment by rthiiyer82** [comment]
This panic issue is fixed in 1.24.1 release.

**Comment by rthiiyer82** [comment]
Please open a new issue if the panic is noticed again. Thanks!
